cocotb/handle.py

```diff
@@ -706,17 +715,30 @@ def _set_value(self, value, call_sim):
         """
         value, set_action = self._check_for_set_action(value)
 
-        if isinstance(value, int) and value < 0x7fffffff and len(self) <= 32:
-            call_sim(self, self._handle.set_signal_val_long, set_action, value)
-            return
+        if isinstance(value, int):
+            if len(self) <= 32:
+                call_sim(self, self._handle.set_signal_val_int, set_action, value, int(_Limits.VECTOR_NBIT))
```

Comment:
This needs to be moved below below the check for overflow. Doing this in C is too late. The traceback will not lead to the site of the bad call.